minkindr/include/kindr/minimal/implementation/rotation-quaternion-inl.h

```diff
@@ -407,8 +409,9 @@ RotationQuaternionTemplate<Scalar>::operator*(
 
   // check if the multiplication has resulted in the quaternion no longer being
   // approximately normalized.
-  if (std::abs(result.squaredNorm() - static_cast<Scalar>(1.0)) >
-      EPS<Scalar>::normalization_value()) {
```

Comment:
This is just a symptom of the issue though, without the normalization it is possible to multiply two valid quaternions to generate an invalid quaternion.
The more multiplications that a quaternion undergoes the more likely it is to happen.